src/cascadia/TerminalSettingsEditor/EditAction.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#pragma once
+
+#include "EditAction.g.h"
+#include "ActionsViewModel.h"
+#include "Utils.h"
+#include "ViewModelHelpers.h"
+
+namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+{
+    struct EditAction : public HasScrollViewer<EditAction>, EditActionT<EditAction>
+    {
+    public:
+        EditAction();
+        void OnNavigatedTo(const winrt::Windows::UI::Xaml::Navigation::NavigationEventArgs& e);
+        void OnNavigatedFrom(const winrt::Windows::UI::Xaml::Navigation::NavigationEventArgs& e);
+
+        til::property_changed_event PropertyChanged;
+
+        WINRT_OBSERVABLE_PROPERTY(Editor::CommandViewModel, ViewModel, PropertyChanged.raise, nullptr);
+
+    private:
+        friend struct EditActionT<EditAction>; // for Xaml to bind events
+        winrt::Windows::UI::Xaml::FrameworkElement::LayoutUpdated_revoker _layoutUpdatedRevoker;
+        winrt::event_token _propagateWindowRootToken{};

```

Comment:
(nit nit: you could also use auto_revoke events like we do for layoutUpdated, but they are more strongly typed. you would need to `.reset()` them instead of calling the registration function a second time to unregister)